### ISSUE TITLE
"New Posts" Indicator for Category with Sub-Forums are not working properly

### ISSUE THREAD

**Issue by euantorano** [mybb/mybb#3398]
I really want this to work, as it is a key component for my forum.  My layout has many categories with different subforums; I would like to show members when there is a new post.  This is usually determined by the "lightbulb", but doesn't seem to work properly

I can give an example of this on here (the myBB support forum); the Themes Category has 3 subforums, and on inspection, the ball shows new posts.  If I read all the posts in the subsections, it still shows that there are new posts

Can I fix this on my forum, or does there need to be an update with MyBB?

Original thread: ["New Posts" Indicator for Category with Sub-Forums are not working properly](https://community.mybb.com/thread-219314.html)

**Comment by P2PUnderground** [comment]
Hi - NotAdmin here :)

I did some quick testing for you because I love you guys.  I don't think the issue is Cache related; it seems to be a code issue

Related to this issue, I found the following when testing

- Pressing the lightbulb doesn't clear the subboards
- If I post in the sub-boards, the lightbulb thinks that I have a new post (i.e. it alerts me to the post I have just made

**Comment by effone** [comment]
Isn't this or related issues bothering us since last 3 years?

https://github.com/mybb/mybb/issues/2041
https://github.com/mybb/mybb/issues/2127

**Comment by euantorano** [comment]
Yes, this has been causing problems for a long time now.

**Comment by Eldenroot** [comment]
@effone - yes, you are right :( maybe there are few more issues which have been there for a long time. I hope they will be sort out in one of later MyBB 1.8.x releases or in MyBB 1.9/1.10.

**Comment by GasmoN** [comment]
Any news about this?

**Comment by euantorano** [comment]
Nothing at the moment. At this point it's probably too late for 1.8.23, but I'll assign this to 1.8.24 to have a look at.

**Comment by GasmoN** [comment]
That would be awesome. I noticed that people are always marking my whole forum as read, they don't even browse new posts that much because of that so all my important news need to be published in live chat.

**Comment by Ben-MyBB** [comment]
Shouldn't the setting "Attempt to Mark Parent Forums as Read" under "Forum Display Options" cover this? @euantorano

**Comment by euantorano** [comment]
Yeah, it might but I need to test. I'd completely forgotten about that setting to be honest!

**Comment by Mikko-B** [comment]
Any news on this? I have the same problem with my forum using version 18.8.30.
I also tried the "Attempt to Mark Parent Forums as Read", but that does not seem to make a difference. Still the problem as described above persists.

**Comment by lazy-val** [comment]
@Mikko-B 
I am trying to reproduce the issue on v18.8.30, but when I use "Attempt to Mark Parent Forums as Read", I don't get the issue.
can you help me to reproduce it or tell me how you get it?

**Comment by Mikko-B** [comment]
@lazy-val-b Sure, I can try to help. You can take a look at https://kaikki.de (you can also create a test account, if you llike).
There are black circles, if they section should contain new posts. When I click on the black circle, this works.
However, when you go to an unread thread (this is marked in bold letters), the thread is marked read and the letters or not bold any more
<img width="226" alt="screen1" src="https://user-images.githubusercontent.com/104639985/178100639-3243fc00-4a6f-4c45-b1c9-4c0f84aae7d7.png">,
but the black circle still shows as if there are unread threads
<img width="145" alt="screen2" src="https://user-images.githubusercontent.com/104639985/178100642-29f23010-3fc2-49b3-9aa1-b7000e79007f.png">
.

**Comment by Stefan-MyBB** [comment]
As guest this works perfectly fine. If this happens for you even without sub-forums it's not related to this bug report and you should ask for support instead.

**Comment by Mikko-B** [comment]
What do you mean "even with sub-forums"? "Sandkasten" is a sub-forum, in which all Threads are read and not marked as containing new posts. Nevertheless, the black circle remains, denoting that there are unread posts.

By the way, in the admin interface there is a comment next to "Attempt to Mark Parent Forums as Read", which states: 
"See the [MyBB Docs](https://docs.mybb.com/versions/1.6.5/) for more information regarding this change."
linking to 1.6.5, even though I'm running 18.8.30. You might want to update that link ;-)

**Comment by Stefan-MyBB** [comment]
"Sandkasten" is forum and has no (visible) sub-forums. It's highly unlikely there is a bug affecting forums without being noticed for years.
